collections.go

```diff
@@ -0,0 +1,121 @@
+package httpshopify
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/MOHC-LTD/httpshopify/internal/http"
+	"github.com/MOHC-LTD/shopify"
+)
+
+type CollectionRepository struct {
+	client    http.Client
+	createURL func(endpoint string) string
+}
+
+func newCollectionRepository(client http.Client, createURL func(endpoint string) string) CollectionRepository {
+	return CollectionRepository{
+		client,
+		createURL,
+	}
+}
+
+func (repository CollectionRepository) Get(ID int64) (shopify.Collection, error) {
+
+	url := repository.createURL(fmt.Sprintf("collections/%v.json%v", ID, parseCollectionQuery()))
+
+	body, _, err := repository.client.Get(url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var resultDTO struct {
+		Collection CollectionDTO `json:"collection"`
+	}
+
+	json.Unmarshal(body, &resultDTO)
+
+	return resultDTO.Collection.ToShopify(), nil
+}
+
+func (repository CollectionRepository) Products(ID int64) (shopify.Products, error) {
+
+	url := repository.createURL(fmt.Sprintf("collections/%v/products.json%v", ID, parseCollectionQuery()))
```

Comment:
Looks like this endpoint is paginated so you'll need to handle that, also dont need the query